Sitefinity CLI/PackageManagement/NuGetApiClient.cs

```diff
@@ -33,26 +33,54 @@ public async Task<NuGetPackage> GetPackageWithFullDependencyTree(string id, stri
             if (File.Exists(packageDependenciesHashFilePath))
                 return JsonConvert.DeserializeObject<NuGetPackage>(File.ReadAllText(packageDependenciesHashFilePath));
 
-            XDocument nuGetPackageXmlDoc = await this.GetPackageXmlDocument(id, version, sources);
+            this.isV2 = true;
+            XDocument nuGetPackageXmlDoc = await this.GetPackageXmlDocument(id, version, sources, httpClient);
             if (nuGetPackageXmlDoc == null)
             {
                 return null;
             }
 
-            XElement propertiesElement = nuGetPackageXmlDoc
-                .Element(this.xmlns + Constants.EntryElem)
-                .Element(this.xmlnsm + Constants.PropertiesElem);
-
+            IList<NuGetPackage> dependencies = new List<NuGetPackage>();
             NuGetPackage nuGetPackage = new NuGetPackage();
-            nuGetPackage.Id = nuGetPackageXmlDoc
-                .Element(this.xmlns + Constants.EntryElem)
-                .Element(this.xmlns + Constants.TitleElem).Value;
+            if (this.isV2)
+            {
+                XElement propertiesElement = nuGetPackageXmlDoc
+                    .Element(this.xmlns + Constants.EntryElem)
+                    .Element(this.xmlnsm + Constants.PropertiesElem);
+
+                nuGetPackage.Id = nuGetPackageXmlDoc
+                    .Element(this.xmlns + Constants.EntryElem)
+                    .Element(this.xmlns + Constants.TitleElem).Value;
 
-            nuGetPackage.Version = propertiesElement.Element(this.xmlnsd + Constants.VersionElem).Value;
+                nuGetPackage.Version = propertiesElement.Element(this.xmlnsd + Constants.VersionElem).Value;
 
-            string dependenciesString = propertiesElement.Element(this.xmlnsd + Constants.DependenciesElem).Value;
+                string dependenciesString = propertiesElement.Element(this.xmlnsd + Constants.DependenciesElem).Value;
+
+                dependencies = this.ParseDependencies(dependenciesString);
+            }
+            else
+            {
+                var packageNamespace = nuGetPackageXmlDoc.Root.GetDefaultNamespace();
+                var elementPackage = nuGetPackageXmlDoc.Element(packageNamespace + Constants.PackageElem);
+                var metadataNamespace = elementPackage.Descendants().First().GetDefaultNamespace();
+                var elementsMetadata = elementPackage.Element(metadataNamespace + Constants.MetadataElem);
+                var groupElementsDependencies = elementsMetadata.Element(metadataNamespace + Constants.DependenciesEl);
 
-            IList<NuGetPackage> dependencies = this.ParseDependencies(dependenciesString);
+                if (groupElementsDependencies != null)
+                {
+                    var v3Regex = new Regex("^.[A-Z]{4}[a-z]{8}\\d.\\d\\b[.0-9]*$", RegexOptions.Compiled);
```

Comment:
It is not clear what v3regex is? Think of a better name

Reply:
the regex is removed